### PR TITLE
Fix app template

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:
- deploys the alloy app as a component named observability-gateway on MCs
- The configuration options are a bit funky because the goal is to use the alloy-app from collections (with the name alloy-gateway) but our internal tooling currently prevents us from deploying an app in a collection with a different name https://github.com/giantswarm/roadmap/issues/3682 is fixed so using this workaround will allow us to switch from this app to the alloy app without having to change the config when we do the migration.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Comment by QuantumEnigmaa** [comment]
I thought the idea was to integrate this app in another repo until we could put it back in collections under the `alloy-gateway` name pointing towards the `alloy-app` repo ? :thinking: 

Otherwise I'm also fine with this method

**Comment by QuentinBisson** [comment]
Well yes this is the other repo :D

**Comment by QuantumEnigmaa** [comment]
I meant "integrate it in another **existing** repo" :sweat_smile:

**Comment by QuentinBisson** [comment]
ah yes but if we did that then we would need to migrate the config :D

**Review by QuantumEnigmaa — APPROVED** [review]
Fine with me

File: helm/alloy-gateway/templates/_helpers.tpl [hunk 1]
```diff
@@ -13,25 +13,14 @@ Create chart name and version as used by the chart label.
...
 Common labels
 */}}
 {{- define "labels.common" -}}
-{{ include "labels.selector" . }}
+app.kubernetes.io/name: {{ include "name" . | quote }}
+app.kubernetes.io/instance: {{ .Release.Name | quote }}
 app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
-application.giantswarm.io/branch: {{ .Chart.Annotations.branch | replace "#" "-" | replace "/" "-" | replace "." "-" | trunc 63 | trimSuffix "-" | quote }}
-application.giantswarm.io/commit: {{ .Chart.Annotations.commit | quote }}
```
Comment (hervenicol) [inline helm/alloy-gateway/templates/_helpers.tpl:32]: We're removing a lot of stuff from the default template. Isn't it good practice to have them?

**Comment by hervenicol** [comment]
It looks like a bit more than a template fix :sweat_smile: 

It's quite hard to spot the differences between generated files and actual code/config.

I haven't seen any config that would make this app act as a Otel / Prometheus / Loki gateway. Did I miss something?

What happens when this is released? As it's pushed to collections it will be deployed to all CAPI MCs. Will it run or will it fail? Are there some related PRs for config?

Also, could you link the source issue?

File: helm/alloy-gateway/templates/_helpers.tpl (hunk 1 above)
Comment (QuentinBisson) [reply helm/alloy-gateway/templates/_helpers.tpl:32]: Commit and branch? We remove it from everywhere actually 😅

**Comment by QuentinBisson** [comment]
So I forgot to link the main issue https://github.com/giantswarm/roadmap/issues/3568#issuecomment-2374031576 ... sorry about this.

This application is deployed via collections so the config (to be) will be defined here https://github.com/giantswarm/shared-configs/blob/main/default/apps/alloy-gateway/configmap-values.yaml.template.

When this app is released it will then be deployed to all CAPI MCs and it will work because I've tested it on grizzly.

Once this is setup, we will be able to configure this properly

**Comment by QuentinBisson** [comment]
@hervenicol let me know if you need more info or we can pair on it

**Comment by hervenicol** [comment]
All right, by default it's not enabled and won't deploy any configmap or pod.
So it's looking good to me.

**Review by hervenicol — APPROVED** [review]
LGTM

**Comment by QuentinBisson** [comment]
Thanks, sorry about the confusion :(

**Pull request merged**